src/NuGet.Frameworks/DefaultFrameworkMappings.cs

```diff
@@ -333,53 +317,115 @@ public IEnumerable<OneWayCompatibilityMappingEntry> CompatibilityMappings
-                                new FrameworkRange(
-                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Silverlight, FrameworkConstants.Version5),
-                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.Silverlight, FrameworkConstants.MaxVersion),
-                                    includeMin: false,
-                                    includeMax: true),
-                                FrameworkConstants.DotNetAll),
-
-                            // wp8 and up support dotnet
-                            new OneWayCompatibilityMappingEntry(
-                                new FrameworkRange(
-                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WindowsPhone, new Version(8, 0, 0, 0)),
-                                    new NuGetFramework(FrameworkConstants.FrameworkIdentifiers.WindowsPhone, FrameworkConstants.MaxVersion)),
-                                FrameworkConstants.DotNetAll),
-
                            // the below frameworks support dotnet
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.DnxCore),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.MonoAndroid),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.MonoMac),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.MonoTouch),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.NetCore),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.UAP),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.Windows),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.WindowsPhoneApp),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinIOs),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinMac),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinPlayStation3),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinPlayStation4),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinPlayStationVita),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinXbox360),
-                           CreateDotNetMappingForAllVersions(FrameworkConstants.FrameworkIdentifiers.XamarinXboxOne),
+                           // dnxcore50 -> dotnet5.3
+                           CreateDotNetGenerationMappingForAllVersions(
```

Comment:
It would need an additional mapping. If dnxcore versions and dot generation mappings are going to be tied together in a predictable way now and forever I could add something to figure it out :)